DIPs/DIP1005.md

```diff
@@ -0,0 +1,488 @@
+
+#### Syntactic Alternatives
+
+There are a number of alternative approaches that have been (and some still are) considered.
+
+* Specify `import` in a manner reminiscent of attributes:
+
+```d
+    void process(File input) import (std.stdio);
+    struct Buffered(import std.range)(Range) if (isInputRange!Range)
+    {
+        ...
+    }
+```
+    This form had significant differences from both the property syntax and the existing `import` syntax.
+
+* Add syntax to allow for an optional `import` declaration inside declarations:
+
+```d
+    void process(import std.stdio)(File input);
+    struct Buffered(import std.range)(Range) if (isInputRange!Range)
+    {
+        ...
+    }
+```
+This has the advantage of being less verbose in case the same module is looked up several times. The disadvantages are a heavier and more ambiguous syntax (two sets of parens for nontemplates, three for templates) and an unclear relationship between the `import`ed entities and the symbols used in the declaration.
+* Use `import` as a pseudo-package such that symbols are written like this:
+
+```d
+        void process(import.std.stdio.File input);
+        struct Buffered(Range) if (import.std.range.isInputRange!Range)
+        {
+            ...
+        }
+```
+Such an option has an ambiguity problem shown by Timon Gehr: is `import.std.range.isInputRange` looking up symbol `isInputRange` in module/package `std.range`, or the symbol `range.isInputRange` (e.g. a `struct` member) in module/package `std`?
+* Stay as close to the existing `import` syntax as possible. This has the advantage of being instantly recognized, but the disadvantage of looking out of place within the declaration:
+
+```d
+        void process(File input) import std.stdio;
+        struct Buffered(Range) if (isInputRange!Range)
+        import std.range;
+        {
+            ...
+        }
+```
+One syntactical issue is that in this case the semicolon ending the `import` may or may not end the declaration; the scanner (and human reader) would need to look ahead to figure whether a definition continues (by means of an open brace, the `in` keyword, or the `out` keyword), or the declaration ends there.
+* Alternatively, the semicolon might be omitted in the approach above. This causes no syntactical ambiguity but makes the hanging `import` declaration even more out of place:
+```d
+        void process(File input) import std.stdio;
+        struct Buffered(Range) if (isInputRange!Range)
+        import std.range
+        {
+            ...
+        }
+```
+* The `with` keyword has been proposed:
+```d
+        void process(File input) import std.stdio;
```

Comment:
I guess this is a copy/paste error and as `with(import std.stdio)` is used below.